src/components/common/form/message-item/MessageItem.jsx

```diff
@@ -1,22 +1,24 @@
 import DatesMoved from '../message-item-types/DatesMoved';
 import { messageKeys } from '../../../../utils/constants';
-import { useTimecardContext } from '../../../../context/TimecardContext';
+import PropTypes from 'prop-types';
 
-const components = {
-  datesMoved: DatesMoved,
-};
-
-const MessageItem = () => {
-  const { summaryMessages } = useTimecardContext();
-  if (messageKeys.update in summaryMessages) {
-    const template = summaryMessages[messageKeys.update].template;
-    const variables = summaryMessages[messageKeys.update].variables;
+const MessageItem = ({ message, setSummaryMessages }) => {
+  if (messageKeys.datesMoved === message.template) {
+    const variables = message.variables;
 
-    const Component = components[template];
-    return <Component variables={variables} />;
+    return (
+      <DatesMoved
+        variables={variables}
+        setSummaryMessages={setSummaryMessages}
+      />
+    );
   } else {
     return null;
   }
 };
 
 export default MessageItem;
+MessageItem.propTypes = {
+  message: PropTypes.object,
```

Comment:
Might be good to use PropTypes.shape to define what the message object should look like

Reply:
Updated to use shape prop type

Reply:
In the corresponding MessageItem tests it looks like there's a `message.key` - not sure if that needs removing from the tests or adding here?